### PR TITLE
avoid fetching even the first set in case cancelled

### PR DESCRIPTION
**Pull request by mkangia**
Just a follow up to https://github.com/dimagi/commcare-hq/pull/27470/files

##### SUMMARY
For a cancelled rule, the task can avoid doing any iteration.
This is not true for a non-sharded run where the helper is initiated right before the iteration itself

##### FEATURE FLAG
`SHARDED_RUN_MESSAGING_RULE`

### CODE REVIEW

File: corehq/messaging/tasks.py [hunk 1]
```diff
@@ -191,11 +191,12 @@ def run_messaging_rule_for_shard(domain, rule_id, db_alias):
 
     chunk_size = getattr(settings, 'MESSAGING_RULE_CASE_CHUNK_SIZE', 100)
     progress_helper = MessagingRuleProgressHelper(rule_id)
-    for case_id_chunk in chunked(paginated_case_ids(domain, rule.case_type, db_alias), chunk_size):
-        sync_case_chunk_for_messaging_rule.delay(domain, case_id_chunk, rule_id)
-        progress_helper.increase_total_case_count(len(case_id_chunk))
-        if progress_helper.is_canceled():
-            break
+    if not progress_helper.is_canceled():
```
Comment (sravfeyn) [inline corehq/messaging/tasks.py:194]: Can we not simply return after this line instead of tabbing following lines?

Comment (mkangia) [reply corehq/messaging/tasks.py:194]: we would still need to update that the shard is complete which happens later.

Comment (sravfeyn) [reply corehq/messaging/tasks.py:194]: 👍

Comment (mkangia) [reply corehq/messaging/tasks.py:194]: cool, we good?

**Review by millerdev — APPROVED** [review]

**Pull request merged**